### ISSUE TITLE
New Products Blocks > filter by cat > wrong data retrieved for Preview/Published (ok in site editor)

### ISSUE THREAD

**Issue by Marc-pi** [woocommerce/woocommerce-gutenberg-products-block#7867]
## Describe the bug

With the new Products Blocks (ex Query Loop)
Filtering by one category only works only in site editor
Preview and Published version are wrong : filter is not working, all products from all cats are displayed

## To reproduce

1. Create products linked to several categories that are not child of each others (only top parent categories)
2. Add the products block within a new WP Page
3. Add Taxonomy Category filter, set on 1 category only
4. Observe result within site editor vs preview vs published

## Expected behavior

The 3 pages status should retrieve the same product list (e.g retrieve the same content)
Here only the site editor version is ok. 
> Category filter does not work on preview and published page.

## Environment

WP6.1.1
Woo Blocks 9.1
Gutenberg plugin 14.6.1
WC 7.1
TT2 Theme

**Comment by imanish003** [comment]
I confirm that issue is reproducible.

**Comment by Marc-pi** [comment]
ping, the issue is still there with Woo Blocks 9.5 !!!! And it is a major one
The use case : add products query block, set with a filter in block settings on 1 category
What's wrong :  
   - currently Editor view is wrong (all products from all cats are displayed)
   - preview and published are ok (only products from the selected cat in block settings are displayed)

Seems someone tried to fix, but reverted the previous buggy behavior !!

![image](https://user-images.githubusercontent.com/919332/216081309-2e9f0891-33e0-4a51-8e8f-21017511f7cd.png)

**Comment by imanish003** [comment]
Hey @Marc-pi,

Thank you for bringing this to my attention. 🙌🏻

After conducting a thorough investigation, I have found that the problem lies within the "Query Loop" block in the Gutenberg editor. As a result, the issue is also present in the "Products" block, which is a variation of the "Query Loop" block. I have discovered that there is already an open issue regarding this on the Gutenberg repository (https://github.com/WordPress/gutenberg/issues/29438), and I have added my observations as a comment for reference.
